components/eamxx/src/share/field/field_impl.hpp

```diff
@@ -143,66 +142,80 @@ auto Field::get_view () const
   return DstView(view_ND);
 }
 
-template<typename DT, HostOrDevice HD>
-auto Field::get_strided_view () const
- -> get_strided_view_type<DT,HD>
-{
+template <typename DT, HostOrDevice HD>
+auto Field::get_strided_view() const ->
+get_strided_view_type<DT, HD> {
   // The destination view type on correct mem space
-  using DstView = get_strided_view_type<DT,HD>;
+  using DstView = get_strided_view_type<DT, HD>;
   // The dst value types
   using DstValueType = typename DstView::traits::value_type;
   // We only allow to reshape to a view of the correct rank
   constexpr int DstRank = DstView::rank;
-  constexpr int DstRankDynamic= DstView::rank_dynamic;
+  constexpr int DstRankDynamic = DstView::rank_dynamic;
 
   // Get src details
   const auto& alloc_prop = m_header->get_alloc_properties();
   const auto& fl = m_header->get_identifier().get_layout();
 
   // Checks
-  EKAT_REQUIRE_MSG (DstRank==1 && fl.rank()==1,
-      "Error! Strided view only available for rank-1 fields.\n");
-  EKAT_REQUIRE_MSG (DstRankDynamic==1,
-      "Error! Strided view not allowed with compile-time dimensions.\n");
-  EKAT_REQUIRE_MSG(is_allocated(),
+  EKAT_REQUIRE_MSG(
+      is_allocated(),
       "Error! Cannot extract a field's view before allocation happens.\n");
-  EKAT_REQUIRE_MSG (not m_is_read_only || std::is_const<DstValueType>::value,
-      "Error! Cannot get a view to non-const data if the field is read-only.\n");
+  EKAT_REQUIRE_MSG(not m_is_read_only || std::is_const<DstValueType>::value,
+                   "Error! Cannot get a view to non-const data if the field is "
+                   "read-only.\n");
   EKAT_REQUIRE_MSG(alloc_prop.template is_compatible<DstValueType>(),
-      "Error! Source field allocation is not compatible with the requested value type.\n");
+                   "Error! Source field allocation is not compatible with the "
+                   "requested value type.\n");
 
   // Check if this field is a subview of another field
   const auto parent = m_header->get_parent().lock();
-  if (parent!=nullptr) {
+  if (parent != nullptr) {
     // Parent field has correct layout to reinterpret the view into N+1-dim view
-    // So create the parent field on the fly, use it to get the N+1-dim view, then subview it.
-    // NOTE: we can set protected members, since f is the same type of this class.
+    // So create the parent field on the fly, use it to get the N+1-dim view,
```

Comment:
This comment should be updated: the parent view may not be N+1 dim, if we are taking multiple slices, but it will have rank N as well (just potentially a bigger extent).

Reply:
Thanks for the catch!